thunder/tests/test_auto_register_torchops.py

```diff
@@ -0,0 +1,197 @@
+            gen = islice(
+                op_info.sample_inputs_func(
+                    op_info, device=torch.device(device), dtype=torch.float32, requires_grad=requires_grad
+                ),
+                5,
+            )
+            for sample in gen:
+                try:
+                    jfun = thunder.jit(func)
+                    out = jfun(sample.input, *sample.args, **sample.kwargs)
+                except Exception as e:
+                    cnt += 1
+                    assert isinstance(e, NotImplementedError)
+                    assert str(e).startswith(f"Exception encountered when doing automatic registration") or str(
+                        e
+                    ).startswith(f"Unsupported type:")
+                    break
+                else:
+                    if requires_grad:
+                        trc = thunder.last_backward_traces(jfun)[-1]
+                        fwd_trc = thunder.last_traces(jfun)[-1]
+                        # skip if it is not differentiable
+                        outs = fwd_trc.output[0]["output"]
+                        outs = outs if isinstance(outs, tuple) else (outs,)
+                        if all(not thunder.core.dtypes.is_inexact_dtype(o.dtype) for o in outs):
+                            continue
+                        vjp_op_name = f"{op_info.name.split('.')[-1]}_vjp"
+                        if op_info.name == "mm":
+                            assert any(bsym.sym.name.endswith(vjp_op_name) for bsym in trc.bound_symbols)
+                        else:
+                            assert any(bsym.sym.name == vjp_op_name for bsym in trc.bound_symbols)
+                    else:
+                        fwd_trc = thunder.last_traces(jfun)[-1]
+                        assert any(
+                            bsym.sym.name.endswith(op_info.name.split(".")[-1]) and not bsym.subsymbols
+                            for bsym in fwd_trc.bound_symbols
+                        )
+            else:
+                suc += 1
+
+    print(f"total number of ops with sample input: {total}, success: {suc}, needs manual registeration: {cnt}")
+
+
+# Replace manual registration of some operations with automatic registration for network test cases
+skip_ops_nanogpt = [
+    get_opinfo("layer_norm"),
+    get_opinfo("linear"),
+    get_opinfo("gelu"),
+    get_opinfo("scaled_dot_product_attention"),
+]
+skip_ops_alexnet = [
+    get_opinfo("conv2d"),
+    get_opinfo("linear"),
+    get_opinfo("adaptive_avg_pool2d"),
+    get_opinfo("max_pool2d"),
+]
+disable_opinfos = skip_ops_nanogpt + skip_ops_alexnet
+tmp1 = dict(thunder.core.jit_ext._general_jit_lookaside_map)
+list(tmp1.pop(k.torch_reference, None) for k in disable_opinfos)
+tmp2 = dict(thunder.torch._torch_to_thunder_function_map)
+list(tmp2.pop(k.torch_reference, None) for k in disable_opinfos)
+tmp3 = dict(thunder.core.jit_ext._minimal_lookaside_map)
+list(tmp3.pop(k.torch_reference, None) for k in disable_opinfos)
```

Comment:
could you tell me what these `tmp` variables and list comprehensions are doing?

Reply:
This is for mocking the ops I want to replace with auto registration, the ops are registered in these global maps, so I tried to remove them in the tmp maps and mock the original ones with the tmp maps.